cmd/gardener-scheduler/app/gardener_scheduler.go

```diff
@@ -0,0 +1,369 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/client-go/discovery"
+	"os"
+	"time"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	gardencoreinformers "github.com/gardener/gardener/pkg/client/core/informers/externalversions"
+	gardeninformers "github.com/gardener/gardener/pkg/client/garden/informers/externalversions"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	gardenmetrics "github.com/gardener/gardener/pkg/controllermanager/metrics"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/scheduler/apis/config"
+	schedulerconfigv1alpha1 "github.com/gardener/gardener/pkg/scheduler/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/scheduler/controller"
+	"github.com/gardener/gardener/pkg/scheduler/validation"
+	"github.com/gardener/gardener/pkg/server"
+	"github.com/gardener/gardener/pkg/server/handlers"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	diskcache "k8s.io/client-go/discovery/cached/disk"
+	"k8s.io/client-go/informers"
+	kubeinformers "k8s.io/client-go/informers"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
+	"github.com/gardener/gardener/cmd/utils"
+)
+
+// Options has all the context and parameters needed to run a GardenerScheduler.
+type Options struct {
+	// ConfigFile is the location of the GardenerScheduler's configuration file.
+	ConfigFile string
+	config     *config.SchedulerConfiguration
+	codecs     serializer.CodecFactory
+}
+
+// AddFlags adds flags for a specific Scheduler to the specified FlagSet.
+func (o *Options) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.ConfigFile, "config", o.ConfigFile, "The path to the configuration file.")
+}
+
+// NewOptions returns a new Options object.
+func NewOptions() (*Options, error) {
+	o := &Options{
+		config: new(config.SchedulerConfiguration),
+	}
+
+	o.codecs = serializer.NewCodecFactory(scheme.Scheme)
+
+	if err := config.AddToScheme(scheme.Scheme); err != nil {
+		return nil, err
+	}
+	if err := schedulerconfigv1alpha1.AddToScheme(scheme.Scheme); err != nil {
+		return nil, err
+	}
+	if err := gardenv1beta1.AddToScheme(scheme.Scheme); err != nil {
+		return nil, err
+	}
+
+	return o, nil
+}
+
+// loadConfigFromFile loads the contents of file and decodes it as a
+// GardenerSchedulerConfiguration object.
+func (o *Options) loadConfigFromFile(file string) (*config.SchedulerConfiguration, error) {
+	data, err := ioutil.ReadFile(file)
```

Comment:
Please move out the configuration loading to a separate package with proper scheme initialization. For reference, check how @rfranzke has done it in gardener-extensions: https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/pkg/apis/config/loader/loader.go